content/src/scripts/components/TreeItem/TreeItem.jsx

```diff
@@ -34,14 +36,23 @@ function TreeItem({
     } else {
       setClicked(true);
       const URLDetails = fetchURLDetails();
-      if ("compatibility-mode" in options && options["compatibility-mode"]) {
-        window.location.href = `${window.location.origin}/${
-          URLDetails.dirFormatted
-        }/blob/${URLDetails.branchName}/${path.join("/")}`;
-      } else {
-        window.location.href = `${window.location.origin}/${
-          URLDetails.dirFormatted
-        }/-/blob/${URLDetails.branchName}/${path.join("/")}`;
+      if (isRepositoryShown()) {
+        if ("compatibility-mode" in options && options["compatibility-mode"]) {
+          window.location.href = `${window.location.origin}/${URLDetails.dirFormatted
+            }/blob/${URLDetails.branchName}/${path.join("/")}`;
+        } else {
+          window.location.href = `${window.location.origin}/${URLDetails.dirFormatted
+            }/-/blob/${URLDetails.branchName}/${path.join("/")}`;
+        }
+      } else if (isMergeRequestShown()) {
+        let hash = sha1(path);
+        let mergeRequestId = grabMergeRequestIdFromCurrentUrl();
+        if (!window.location.href.includes('diffs')) {
+          window.location.href = (window.location.origin + "/" + URLDetails.dirFormatted + "/-/merge_requests/" + mergeRequestId + "/diffs#" + hash);
+        } else {
+          window.location.hash = hash;
+          window.location.reload();
```

Comment:
overkill if you ask me but somehow the window did not refresh when changing the hash only